projects/network_tester/src/ui_task.c

```diff
@@ -657,6 +668,62 @@ static void ui_load_menu(void)
             screen[(menu_pos%3)+2][0] = CURSOR_GLYPH;
             xTaskNotifyGive(s_screen_task_handle);
             break;
+        case SET_DL_BAND_MENU:
+            menu_pos_floor = (menu_pos/3) * 3;
+            llabs_dl_band_cfg_t dl_band_cfg;
+            int8_t active_pos;
+
+            strncpy(screen[0], set_dl_band_title, LCD_COLUMNS);
+            strncpy(screen[1], set_dl_band_text[menu_pos_floor], LCD_COLUMNS);
+            strncpy(screen[2], set_dl_band_text[menu_pos_floor+1], LCD_COLUMNS);
+            strncpy(screen[3], set_dl_band_text[menu_pos_floor+2], LCD_COLUMNS);
+            screen[(menu_pos%3)+1][0] = CURSOR_GLYPH;
+
+	    if (ll_dl_band_cfg_get(&dl_band_cfg) != 0)
+	    {
+                if (dl_band_cfg.band_edge_lower == DL_BAN_FCC.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_1;
+		}
+		else if (dl_band_cfg.band_edge_lower == DL_BAN_BRA.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_2;
+		}
+		else if (dl_band_cfg.band_edge_lower == DL_BAN_AUS.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_3;
+		}
+		else if (dl_band_cfg.band_edge_lower == DL_BAN_NZL.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_4;
+		}
+		else if (dl_band_cfg.band_edge_lower == DL_BAN_ETSI.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_5;
+		}
+		else
+		{
+		    active_pos = -1;
+		}
```

Comment:
The only way I could make this a switch statement is if I set up the downlink configurations in an array that corresponded to a typedef enum that corresponds the name with the array index. This is because switch cases must be constant and cannot be any variable name (even if its constant? :-1: ) I don't think that is worth it though.